storage/Storage.h

```diff
@@ -115,6 +115,64 @@ namespace storage
     };
 
 
+    class LuksInfo;
+
+
+    /**
+     * Specialized callbacks with a more generic parameter for LUKS activation.
+     */
+    class ActivateCallbacksLuks : public ActivateCallbacks
+    {
+    public:
+
+	virtual ~ActivateCallbacksLuks() {}
+
+	/**
+	 * Decide whether the LUKS should be activated.
+	 *
+	 * Parameter "info" contains all known information about the LUKS device.
+	 */
+	virtual std::pair<bool, std::string> luks(const LuksInfo& info, int attempt) const = 0;
+
+    };
+
+
+    /**
+     * Stores information about a LUKS device.
+     */
+    class LuksInfo : private boost::noncopyable
+    {
+    public:
+
+	LuksInfo();
+
+	~LuksInfo();
+
+	std::string get_device_name() const;
+
+	std::string get_uuid() const;
+
+	std::string get_label() const;
+
+	void set_device_name(const std::string& device_name);
```

Comment:
These seem not required in the interface.

Reply:
Ok. I have added parameters to the constructor to avoid setters in the public interface.

Reply:
Finally I have adapted it according to @aschnell suggestion (set the values by using the impl object directly). Thanks!